services/dynamic-sidecar/src/simcore_service_dynamic_sidecar/core/rabbitmq.py

```diff
@@ -1,212 +1,83 @@
-from __future__ import annotations
-
-import asyncio
 import logging
-import os
-import socket
-from asyncio import CancelledError, Queue, Task
-from contextlib import suppress
-from typing import Any
+from typing import Union, cast
 
-import aio_pika
 from fastapi import FastAPI
-from models_library.projects import ProjectID
-from models_library.projects_nodes import NodeID
 from models_library.rabbitmq_messages import (
     EventRabbitMessage,
     LoggerRabbitMessage,
     RabbitEventMessageType,
+    RabbitMessageBase,
 )
-from models_library.users import UserID
-from servicelib.rabbitmq_utils import RabbitMQRetryPolicyUponInitialization
-from settings_library.rabbit import RabbitSettings
-from tenacity._asyncio import AsyncRetrying
+from servicelib.logging_utils import log_context
+from servicelib.rabbitmq import RabbitMQClient
+from servicelib.rabbitmq_utils import wait_till_rabbitmq_responsive
 
 from ..core.settings import ApplicationSettings
 
 log = logging.getLogger(__file__)
 
-# limit logs displayed
-logging.getLogger("aio_pika").setLevel(logging.WARNING)
-
 
-SLEEP_BETWEEN_SENDS: float = 1.0
+async def _post_rabbit_message(app: FastAPI, message: RabbitMessageBase) -> None:
+    # NOTE: this check is necessary when the dy-sidecar is used on the CLI
+    # where the rabbit is not initialized, it's not optimal but it allows
+    # to run the CLI without rabbit...
+    if hasattr(app.state, "rabbitmq_client"):
```

Comment:
I would wrap this in a function and cache the result. This call can be called multiple times per second. So some caching could help.

Reply:
I am not very happy with this actually. and this is here just because of the sidecar CLI that cannot cope with not having the rabbitmq... used lru_cache now